src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -850,26 +851,51 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
 
             // bucketize
             // prevent reallocation. it may not have 'count' entries, but it won't have more. 
-            var accumulator = new Dictionary<K, ArrayBuilder<T>>(items.Length, comparer);
+            //
+            // We store a mapping from keys to either a single item (very common in practice as this is used from
+            // callers that maps names to symbols with that name, and most names are unique), or an array builder of items.
+
+            var accumulator =  new Dictionary<K, object>(items.Length, comparer);
```

Comment:
Feels like there would be less suppressions if you make this `object?`?

Reply:
fixed.